bukuserver/server.py

```diff
@@ -154,31 +120,25 @@ def shell_context():
-    app.add_url_rule('/api/bookmarks/<int:rec_id>', view_func=bookmark_api_view, methods=['GET', 'PUT', 'DELETE'])
-    app.add_url_rule('/api/bookmarks/refresh', 'refresh_bookmark', refresh_bookmark, defaults={'rec_id': None}, methods=['POST'])
-    app.add_url_rule('/api/bookmarks/<int:rec_id>/refresh', 'refresh_bookmark', refresh_bookmark, methods=['POST'])
-    app.add_url_rule('/api/bookmarks/<int:rec_id>/tiny', 'get_tiny_url', get_tiny_url, methods=['GET'])
-    app.add_url_rule('/api/network_handle', 'network_handle', handle_network, methods=['POST'])
-    app.add_url_rule('/api/fetch_data', 'fetch_data', fetch_data, methods=['POST'])
-    bookmark_range_api_view = api.ApiBookmarkRangeView.as_view('bookmark_range_api')
-    app.add_url_rule(
-        '/api/bookmarks/<int:starting_id>/<int:ending_id>',
-        view_func=bookmark_range_api_view, methods=['GET', 'PUT', 'DELETE'])
-    bookmark_search_api_view = api.ApiBookmarkSearchView.as_view('bookmark_search_api')
-    app.add_url_rule('/api/bookmarks/search', view_func=bookmark_search_api_view, methods=['GET', 'DELETE'])
-    bookmarklet_view = api.BookmarkletView.as_view('bookmarklet')
-    app.add_url_rule('/bookmarklet', view_func=bookmarklet_view, methods=['GET'])
+    app.add_url_rule('/api/tags', 'get_all_tags', api.get_all_tags, methods=['GET'], strict_slashes=False)
+    app.add_url_rule('/api/tags/<tag>', view_func=api.ApiTagView.as_view('tag'), methods=['GET', 'PUT', 'DELETE'])
+    app.add_url_rule('/api/bookmarks', view_func=api.ApiBookmarksView.as_view('bookmarks'), methods=['GET', 'POST', 'DELETE'])
+    app.add_url_rule('/api/bookmarks/<int:index>', view_func=api.ApiBookmarkView.as_view('bookmark'), methods=['GET', 'PUT', 'DELETE'])
+    app.add_url_rule('/api/bookmarks/refresh', 'bookmarks_refresh', api.refresh_bookmark, defaults={'index': None}, methods=['POST'])
+    app.add_url_rule('/api/bookmarks/<int:index>/refresh', 'bookmark_refresh', api.refresh_bookmark, methods=['POST'])
+    app.add_url_rule('/api/bookmarks/<int:index>/tiny', 'tiny_url', api.get_tiny_url, methods=['GET'])
+    app.add_url_rule('/api/bookmarks/<int:start_index>/<int:end_index>',
+                     view_func=api.ApiBookmarkRangeView.as_view('bookmark_range'), methods=['GET', 'PUT', 'DELETE'])
+    app.add_url_rule('/api/bookmarks/search', view_func=api.ApiBookmarkSearchView.as_view('bookmarks_search'), methods=['GET', 'DELETE'])
+    app.add_url_rule('/api/network_handle', 'network_handle', api.handle_network, methods=['POST'])
+    app.add_url_rule('/api/fetch_data', 'fetch_data', api.fetch_data, methods=['POST'])
 
     #  non api
     @app.route('/favicon.ico')
     def favicon():
         return redirect(url_for('static', filename='bukuserver/favicon.svg'), code=301)  # permanent redirect
 
+    app.add_url_rule('/bookmarklet', 'bookmarklet', api.bookmarklet_redirect, methods=['GET'])
```

Comment:
This endpoint is not actually a part of API (even though it obtains the DB object as such)